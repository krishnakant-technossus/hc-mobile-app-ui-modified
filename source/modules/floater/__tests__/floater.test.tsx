// npx jest ./source/modules/floater/__tests__/floater.test.ts

import * as React from 'react';;
import {render} from '@testing-library/react-native';

import Floater from '../floater';
import Button from '../../button/button';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Floater', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Floater alignment="bottom">
            <Button>Primary Button</Button>
          </Floater>
        </ThemeProvider>,
      );

      const floater = wrapper.getByTestId('floater');

      expect(floater).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders its children correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Floater alignment="bottom">
            <Button>Primary Button</Button>
          </Floater>
        </ThemeProvider>,
      );

      const button = wrapper.getByTestId('button');

      expect(button).toBeDefined();
    });
  });
});
