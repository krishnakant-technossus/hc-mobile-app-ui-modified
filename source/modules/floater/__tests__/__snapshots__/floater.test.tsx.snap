// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floater Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      },
    ]
  }
  testID="floater"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00AEFF",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderWidth": 0,
        "display": "flex",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      }
    }
    testID="button"
  >
    <Text
      appearance="filled"
      color="primary"
      disabled={false}
      small={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
      testID="button-label"
    >
      Primary Button
    </Text>
  </View>
</View>
`;
